### PR TITLE
[mqtt] ChannelStateTests.java

### LINKED ISSUE #5406: [mqtt] ChannelStateTests.receiveDateTimeTest failing
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.openhab.binding.mqtt.ChannelStateTests
[ERROR] Tests run: 12, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.874 s <<< FAILURE! - in org.openhab.binding.mqtt.ChannelStateTests
[ERROR] receiveDateTimeTest(org.openhab.binding.mqtt.ChannelStateTests)  Time elapsed: 0.672 s  <<< FAILURE!
java.lang.AssertionError:

Expected: is "2019-04-05T20:41:55.077+0100"
     but: was "2019-04-05T20:41:55.077+0200"
        at org.openhab.binding.mqtt.ChannelStateTests.receiveDateTimeTest(ChannelStateTests.java:278)

[INFO] Running org.openhab.binding.mqtt.ChannelStateTransformationTests
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.172 s - in org.openhab.binding.mqtt.ChannelStateTransformationTests
[INFO] Running org.openhab.binding.mqtt.internal.handler.GenericThingHandlerTests
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.016 s - in org.openhab.binding.mqtt.internal.handler.GenericThingHandlerTests
[INFO] Running org.openhab.binding.mqtt.mapping.MqttTopicClassMapperTests
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.078 s - in org.openhab.binding.mqtt.mapping.MqttTopicClassMapperTests
[INFO] Running org.openhab.binding.mqtt.mapping.SubscribeFieldToMQTTtopicTests
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.125 s - in org.openhab.binding.mqtt.mapping.SubscribeFieldToMQTTtopicTests
[INFO] Running org.openhab.binding.mqtt.values.ValueTests
[INFO] Tests run: 13, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in org.openhab.binding.mqtt.values.ValueTests
[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   ChannelStateTests.receiveDateTimeTest:278
Expected: is "2019-04-05T20:41:55.077+0100"
     but: was "2019-04-05T20:41:55.077+0200"
[INFO]
[ERROR] Tests run: 38, Failures: 1, Errors: 0, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:

[INFO] openHAB Add-ons :: Bundles :: MQTT Things and Channels FAILURE [  4.013 s]

[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project org.openhab.binding.mqtt.generic: There are test failures.
```

### PR DESCRIPTION
**Pull request by martinvw**
Some cleanups & made test independent of time zones

Closes #5406

### CODE REVIEW

File: bundles/org.openhab.binding.mqtt.generic/src/test/java/org/openhab/binding/mqtt/generic/ChannelStateTests.java [hunk 1]
```diff
@@ -211,85 +208,89 @@ public void receivePercentageTest() throws InterruptedException, ExecutionExcept
...
         ZonedDateTime zd = ZonedDateTime.now();
         String datetime = zd.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
-        c.processMessage("state", datetime.getBytes());
-        assertThat(value.getChannelState().toString(), is(datetime + "+0100"));
-        assertThat(value.getMQTTpublishValue().toString(), is(datetime));
+
+        subject.processMessage("state", datetime.getBytes());
+
+        String channelState = value.getChannelState().toString();
+        assertTrue("Expected '" + channelState + "' to start with '" + datetime + "'",
```
Comment (J-N-K) [inline bundles/org.openhab.binding.mqtt.generic/src/test/java/org/openhab/binding/mqtt/generic/ChannelStateTests.java]: I have been told in the Onewire binding (in ESH) not to use this pattern (additional message in asserts), Has this changed? I personally think that this makes a lot tests more clear.

Comment (davidgraeff) [reply bundles/org.openhab.binding.mqtt.generic/src/test/java/org/openhab/binding/mqtt/generic/ChannelStateTests.java]: I guess there was a way to use `assertThat` in your case which is preferred I think. But I'm happy with `assertTrue` as well in this test.

Comment (martinvw) [reply bundles/org.openhab.binding.mqtt.generic/src/test/java/org/openhab/binding/mqtt/generic/ChannelStateTests.java]: @j-n-k do you have a link to this I’m curious about the why?

Comment (J-N-K) [reply bundles/org.openhab.binding.mqtt.generic/src/test/java/org/openhab/binding/mqtt/generic/ChannelStateTests.java]: Unfortunately I can't find that comment anymore. It seems that it wasn't in the initial contribution but at a later point. Could also have been in the DMX binding. IIRC the argument was that the test itself is enough description and the additional string just pollutes the code.

**Pull request merged**